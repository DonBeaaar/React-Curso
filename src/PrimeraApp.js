import React from "react";
import PropType from "prop-types";

// Comunicación entre componentes desestructuada
const PrimeraApp = ({saludo, subtitulo}) => {
  return (
    // Fragment corto
    <>
      <h1>{saludo}</h1>
      <p>{subtitulo}</p>
    </>
  );
};

//Definición de tipado de los props
PrimeraApp.propTypes = {
    saludo: PropType.string
}

//Valores por defecto en los componentes
PrimeraApp.defaultProps= {
    subtitulo: "Soy un subtitulo"
}

export default PrimeraApp;
