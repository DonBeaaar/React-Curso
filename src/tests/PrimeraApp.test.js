import React from "react";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas enn PrimeraApp", () => {
  // test('debe mostrar mensaje "Hola mundo"',()=>{

  //     const saludo = 'Hola mundo';
  //     const {getByText} = render(<PrimeraApp saludo={saludo} />);

  //     expect(getByText(saludo)).toBeInTheDocument()
  // });

  test("despliegue correcto de <PrimeraApp />", () => {
    const saludo = "Hola hola";

    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar el subitutlo ennviado por props", () => {
    const saludo = "Hola hola";
    const subTitulo = "Subtitulo";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );

    // Le estoy diciendo que busque el o los elementos html <p/>
    // y que obtenga el texto del mismo
    const subTest = wrapper.find('p').text();

    expect(subTest).toBe(subTitulo);
  });
});
