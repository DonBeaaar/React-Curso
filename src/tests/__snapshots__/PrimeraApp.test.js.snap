// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas enn PrimeraApp despliegue correcto de <PrimeraApp /> 1`] = `
<Fragment>
  <h1>
    Hola hola
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
